docs/reference.md

```diff
@@ -0,0 +1,214 @@
+# Classes
+
+## Client
+
+The `Client` class is the Usine orchestrator. You generally should not interact
+directly with it.
+
+### Constructor arguments
+
+- **hostname**: a `str` representing the host to connect to. It can be a domain,
+  an IP address, an SSH host, a `user@host` string…
+- **configpath**: a filepath (or list of filepaths) to yaml config file(s) to
+  be loaded
+
+## Config
+
+The `Config` class is a key/value proxy. You'll generally use it through the
+`config` singleton.
+
+You can access config properties with the dot syntax:
+
+```python
+from usine import config
+
+print(config.property)
+```
+
+## Status
+
+A command return status. You can print it to get the command output or test it
+in a boolean context to check its exit code.
+
+```python
+from usine import run
+
+status = run('createuser bob')
+if not status:
+    do_something()
+```
+
+# Connection helpers
+
+
+## enter
+
+Creates the `client` singleton, which in turns initiate the SSH connection.
+
+It takes the same arguments as the `Client`, plus an optionnal `client` kwarg
+that you can use to define the `Client` class to be instantiated.
+
+## exit
+
+Closes the SSH connection.
+
+## connect
+
+`connect` is a contextmanager that will call `enter` and `exit` for you.
+
+```
+from usine import connect
+
+with connect(hostname='me@remote'):
+    run('something')
+```
+
+# Command helpers
+
+## run(cmd)
+
+This is the main helper, which basically runs any command on the remote server.
+
+##### Arguments
+
+- **cmd**: the actual command to be run
+
+Returns as `Status` instance.
+
+## exists(path)
+
+Check if a path (file or directory) exists on the remote server.
+
+##### Arguments
+
+- **path**: the path to check
+
+## mkdir(path, parents=True, mode=None)
+
+Run `mkdir` command on the remote server to create a directory. See `man mkdir`
+for more info about the command.
+
+##### Arguments:
+
+- **path**: the path of the directory
+- **parents** (default: True): also create parents directory if needed and do
+  not raise if the directory already exists.
+- **mode**: optionnal file mode
+
+
+## chown(mode, path, recursive=True, preserve_root=True)
+
+Run `chown` command in the remote server to change file or directory ownership.
+See `man chown` for more info.
+
+##### Arguments
+
+- **mode**: `user:group` to changer the ownership to
+- **path**: the path of the file or directory
+- **recursive** (default: True): operate on files and directories recursively
```

Comment:
operateS?

Reply:
Good question :p
It's never clear for me.
This sentence is actually a copy-paste from `man chown`, and there I see they never use the "s".
So maybe I should instead align on that way everywhere?

Reply:
No strong opinion about that, do what is consistent over code and I'm OK :)

Reply:
I've hesitated but eventually removed all the "S".